jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/log/RuntimeLogPresenter.java

```diff
@@ -53,55 +59,83 @@ public IsWidget getWidget() {
         return view;
     }
 
+    public void setCurrentProcessInstanceId(Long currentProcessInstanceId) {
+        this.currentProcessInstanceId = currentProcessInstanceId;
+    }
+
+    public String getCurrentProcessName() {
+        return currentProcessName;
+    }
+
+    public void setCurrentProcessName(String currentProcessName) {
+        this.currentProcessName = currentProcessName;
+    }
+
+    public void setCurrentServerTemplateId(String currentServerTemplateId) {
+        this.currentServerTemplateId = currentServerTemplateId;
+    }
+
+    public void setCurrentDeploymentId(String currentDeploymentId) {
+        this.currentDeploymentId = currentDeploymentId;
+    }
+
     public void refreshProcessInstanceData(final LogOrder logOrder,
                                            final LogType logType) {
+        processRuntimeDataService.call((List<RuntimeLogSummary> logs) -> {
+            final List<String> logsLine = new ArrayList<String>();
+            if (logOrder == LogOrder.ASC) {
+                Collections.reverse(logs);
+            }
+            logs.stream().forEach(rls -> addLogline(logsLine,
+                                                    rls,
+                                                    logType));
+
+            view.setLogs(logsLine);
+        }).getProcessInstanceLogs(currentServerTemplateId,
+                                  currentDeploymentId,
+                                  currentProcessInstanceId);
+    }
 
+    protected void addLogline(List<String> logLines,
+                              RuntimeLogSummary logSummary,
+                              LogType logType) {
         if (LogType.TECHNICAL.equals(logType)) {
-            processRuntimeDataService.call(new RemoteCallback<List<RuntimeLogSummary>>() {
-                @Override
-                public void callback(List<RuntimeLogSummary> logs) {
-                    final List<String> logsLine = new ArrayList<String>();
-
-                    if (logOrder == LogOrder.DESC) {
-                        Collections.reverse(logs);
-                    }
-
-                    for (RuntimeLogSummary rls : logs) {
-                        logsLine.add(rls.getTime() + ": " + rls.getLogLine() + " - " + rls.getType());
-                    }
-
-                    view.setLogs(logsLine);
-                }
-            }).getRuntimeLogs(currentServerTemplateId,
-                              currentDeploymentId,
-                              currentProcessInstanceId);
+            String agent = constants.System();
+            if (NODE_HUMAN_TASK.equals(logSummary.getNodeType()) && logSummary.isCompleted()) {
+                agent = constants.Human();
+            } else if (NODE_START.equals(logSummary.getNodeType()) && !logSummary.isCompleted()) {
+                agent = constants.Human();
+            }
+            logLines.add(logSummary.getPrettyTime() + ": " +
```

Comment:
I think it makes a bit more readable if you use String.format instead of concatenating all these Strings together...